api/provider/aws/common_test.go

```diff
@@ -3,11 +3,86 @@ package aws
 import (
 	"testing"
 
+	"github.com/aws/aws-sdk-go/service/ecs"
 	"github.com/quintilesims/layer0/api/tag"
 	"github.com/quintilesims/layer0/common/models"
 	"github.com/stretchr/testify/assert"
 )
 
+func Test_getLaunchTypeFromEnvironmentID(t *testing.T) {
+	tagStore := tag.NewMemoryStore()
+
+	envIDs := []string{"env_id0", "env_id1"}
+
+	tags := models.Tags{
+		{
+			EntityID:   envIDs[0],
+			EntityType: "environment",
+			Key:        "type",
+			Value:      models.EnvironmentTypeDynamic,
+		},
+		{
+			EntityID:   envIDs[1],
+			EntityType: "environment",
+			Key:        "type",
+			Value:      models.EnvironmentTypeStatic,
+		},
+	}
+
+	for _, tag := range tags {
+		if err := tagStore.Insert(tag); err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	cases := map[string]string{
+		envIDs[0]: ecs.LaunchTypeFargate,
+		envIDs[1]: ecs.LaunchTypeEc2,
+	}
+
+	for id, expected := range cases {
+		result, err := getLaunchTypeFromEnvironmentID(tagStore, id)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		assert.Equal(t, expected, result)
+	}
+}
+
+func Test_getLaunchTypeFromEnvironmentID_Errors(t *testing.T) {
+	tagStore := tag.NewMemoryStore()
+
+	envIDs := []string{"env_id0", "env_id1"}
+
+	tags := models.Tags{
+		{
+			EntityID:   envIDs[0],
+			EntityType: "environment",
+			Key:        "type",
+			Value:      "",
+		},
+		{
+			EntityID:   envIDs[1],
+			EntityType: "environment",
+			Key:        "",
+			Value:      "",
+		},
+	}
+
+	for _, tag := range tags {
+		if err := tagStore.Insert(tag); err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	for _, id := range envIDs {
+		if _, err := getLaunchTypeFromEnvironmentID(tagStore, id); err == nil {
```

Comment:
[nit] we typically run each case in a table test as a subtest using [t.Run](https://golang.org/pkg/testing/#T.Run)

Reply:
✅